### ISSUE TITLE
Omxplayer subtitles and the vc4-fkms-v3d overlay

### ISSUE THREAD

**Issue by vanfanel** [popcornmix/omxplayer#548]
Hi,

Using vc4-vkms-v3d overlay allows omxplayer to work along with KMS/DRM apps (which I program and use), but it seems that omxplayer uses openvg for subtitles so it won't work without disabling osd explicitly (--no-osd option).
Any chances that omxplayer subs work with the vc4-vkms-v3d by simply using dispmanx to display subs on their own dispmanx element or other similar solution?

I really love omxplayer, it's maybe the most incredible piece of software running on the Pi along with the Linux kernel, and I think this deserves to be fixed for the near future...

**Comment by popcornmix** [comment]
Yes, eventually raspberry pi will move to the arm side graphics driver and firmware side GL and VG drivers will be unavailable.

OpenVG hasn't really caught on and I don't think there is a usable arm side OpenVG driver (see [here](https://www.phoronix.com/scan.php?page=news_item&px=gallium3d-openvg-vega-removed)).

So we'll need a new scheme for rendering subtitles. FreeType should be able to render to a bitmap without using OpenVG, and that bitmap can be displayed through dispmanx as before.

**Comment by vanfanel** [comment]
@popcornmix : Do you know how "eventual" could it be? I mean, are there any dates till the first Raspbian version starts booting in KMS mode by default?

**Comment by popcornmix** [comment]
When its ready.

We will be using the fkms driver before the kms driver, and we'll need to ensure all standard raspbian apps are working with the fkms driver before we consider switching

I'd say using fkms is at least a few months away, but probably before the end of the year.

**Comment by vanfanel** [comment]
@popcornmix: shouldn't mpv work with mmal-accelerated ffmpeg under vc4-fkms-v3d or even vc4-kms-v3d? (I don't know if fkms would be needed in this scenario or why)
Why not provide ready-to-use mpv + mmal accelerated ffmpeg instead of omxplayer?

**Comment by popcornmix** [comment]
I've not really used mpv. Does it have the same performance as omxplayer?
Can you play 1080p without frame drops on a Pi1?

**Comment by popcornmix** [comment]
I'd say not now. There is work going on with supporting V4L2 decode, ffmpeg and kodi with hardware video decode using VC4 graphics world but it's currently experimental (requiring custom kernel and patches to ffmpeg). But it is something we are working towards.

**Comment by vanfanel** [comment]
@popcornmix : Thanks for the response. So, is V4L2 the chosen API to implement accelerated video decoding in ffmpeg with the open source graphics stack instead of the previuosly used MMAL?
Also, is omxplayer being deprecated in favour of a "standard" ffmpeg/mpv based solution as the default Raspbian media playing software, once V4L2 acceleration is working?

**Comment by popcornmix** [comment]
Technically V4L2 uses mmal under the bonnet, so it's still involved but V4L2 is a better choice for players that support multiple platforms, as it is a standard (although the playback side is fairly new and doesn't expose everything needed but that will hopefully improve with time). 
 
Assuming we can get the required performance with a standard media player (e.g. VLC/MPV) then that will likely be recommended over omxplayer.
omxplayer is quite basic in features, and getting the GUI/subtitles/network/DVD handled by a standard player is beneficial.
